**Step 4:** Solve the model

The problem is solved using python for over 1000 iterations.


In [1]:
import pandas as pd
import random
import math

In [2]:
# Defining function for z = p(r)
def prob(x, y, a, b):
    if math.sqrt((x-a)**2 + (y-b)**2) <= 60:
        return 1 - (math.sqrt((x-a)**2 + (y-b)**2))/60
    else:
        return 0

In [3]:
# Defining function for Probability average:
def ave(x,y):
    return (prob(x, y, 10, 18) + prob(x, y, 10, 26) + prob(x, y, 10, 36) + prob(x, y, 10, 46) + prob(x, y, 10, 56) 
            + prob(x, y, 10, 68) + prob(x, y, 10, 76) + prob(x, y, 12, 18) + prob(x, y, 12, 22) + prob(x, y, 12, 26) 
            + prob(x, y, 12, 32) + prob(x, y, 26, 18) + prob(x, y, 26, 22) + prob(x, y, 26, 26) + prob(x, y, 26, 32)
            + prob(x, y, 26, 72) + prob(x, y, 34, 20) + prob(x, y, 36, 62) + prob(x, y, 40, 20) + prob(x, y, 48, 24)
            + prob(x, y, 48, 30) + prob(x, y, 48, 36) + prob(x, y, 48, 46) + prob(x, y, 48, 52) + prob(x, y, 48, 58)
            + prob(x, y, 48, 72) + prob(x, y, 52, 12) + prob(x, y, 52, 20) + prob(x, y, 52, 26) + prob(x, y, 52, 34)
            + prob(x, y, 52, 42) + prob(x, y, 52, 50) + prob(x, y, 52, 56)) / 33

In [4]:
# Defining function for Random Search Algorithm
def sim(n):
    # Data frame for results
    z = pd.DataFrame(columns=["x", "y", "probability"])
    
    # Loop for 1st shape
    for i in range(1, n//6 + 1):
        x = random.uniform(10, 12)
        y = random.uniform(10, 72)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    # Loop for 2nd shape
    for i in range(n//6 +1, 2*n//6 + 1):
        x = random.uniform(12, 26)
        y = random.uniform(10, 16)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    # Loop for 3rd shape
    for i in range(2*n//6 + 1, 3*n//6 + 1):
        x = random.uniform(26, 52)
        y = random.uniform(10, 20)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    # Loop for 4th shape
    for i in range(3*n//6 + 1, 4*n//6 + 1):
        x = random.uniform(26, 30)
        y = random.uniform(20, 62)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    # Loop for 5th shape
    for i in range(4*n//6 + 1, 5*n//6 + 1):
        x = random.uniform(48, 52)
        y = random.uniform(20, 62)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    # Loop for 6th shape
    for i in range(5*n//6 + 1, n + 1):
        x = random.uniform(26, 52)
        y = random.uniform(62, 72)
        z.loc[i-1, "x"] = x
        z.loc[i-1, "y"] = y
        z.loc[i-1, "probability"] = ave(x,y)

    result = z[z["probability"] == max(z["probability"])]
    result.columns
    return result

**Step 5:** Answer the Question

In [6]:
sim(1000)

x          y probability
612  29.745105  32.857931    0.599824

Therefore, at **(29.75, 32.86)**, the probability of customers visiting the kiosk is maximized to **0.6**.